README.rst

```diff
@@ -92,138 +92,172 @@ Running Tests
+    dcos cluster setup ${CLI_TEST_DCOS_URL} \
+        --insecure \
+        --username=bootstrapuser \
+        --password=deleteme
+    deactivate
+    cd -
+
+    cp /etc/hosts ${DCOS_DIR}/hosts.local
+    grep -q "^.* dcos.snakeoil.mesosphere.com$" ${DCOS_DIR}/hosts.local && \
+        sed -iold "s/^.* dcos.snakeoil.mesosphere.com$/${CLI_TEST_DCOS_URL} dcos.snakeoil.mesosphere.com/" ${DCOS_DIR}/hosts.local || \
+        echo ${CLI_TEST_DCOS_URL} dcos.snakeoil.mesosphere.com >> ${DCOS_DIR}/hosts.local
+    sudo mv ${DCOS_DIR}/hosts.local /etc/hosts
+
+**CLI_TEST_DCOS_URL**: Holds the URL or IP address of the cluster you
+are testing against. If you used :code:`dcos-launch` to launch the cluster,
+you can get the IP of the cluster by running :code:`dcos-launch describe`.
+
+**CLI_TEST_SSH_KEY_PATH**: Points to a private key file used to ssh into
+nodes on your cluster. If you used :code:`dcos-launch` to launch the cluster,
+then this should point to the same file used in your :code:`dcos-launch`
+config. This is used by the :code:`node` integration tests.
+
+**CLI_TEST_SSH_USER**: Holds the username used to ssh into nodes on your
+cluster. If you used :code:`dcos-launch` with the configuration listed above
+to launch your cluster, then you *must* set this to `centos`. This is used
+by the :code:`node` integration tests.
 
 Running
 #######
 
-There are two ways to run tests, you can either use the virtualenv created by
-:code:`make env` above::
-
-    make test
-
-Or, assuming you have tox installed (via :code:`sudo pip install tox`)::
-
-    tox
-
-Either way, tox will run unit and integration tests in Python 3.5 using a
-temporarily created virtualenv.
-
-*NOTE:* In order for all the integration tests to pass, your DC/OS cluster must
-have the experimental packaging features enabled. In order to enable these
-features the :code:`staged_package_storage_uri` and :code:`package_storage_uri`
-configuration paramenters must be set at cluster setup.  See `dcos
-configuration parameters`_ for more information.
+Now that your environment is set up appropriately, we can start running the
+tests. We have tests both in the :code:`dcos` package (root directory)
+and in the :code:`dcoscli` package (:code:`cli` directory).
 
-The easiest way to launch a cluster with these capabilities is to use
-`dcos-launch`_ with the configuration listed below::
+When running the tests, change your current directory to one of those two
+locations and follow the instructions below.
 
-    launch_config_version: 1
-    deployment_name: ${DEPLOYMENT_NAME}
-    template_url: ${TEMPLATE_URL}
-    provider: aws
-    aws_region: us-west-2
-    template_parameters:
-        KeyName: default
-        AdminLocation: 0.0.0.0/0
-        PublicSlaveInstanceCount: 1
-        SlaveInstanceCount: 1
+*NOTE*: You **must** have your virtualenv *deactivated* in order to run the
+tests via the commands below. This is very important and often a point of
+much confusion.
```

Comment:
Just curious, why is that?

Reply:
@klueska : I had forgot about this one, could you explain why? How would integration tests will find the dcos executable for the current code?